staging/src/k8s.io/apiserver/pkg/util/flag/langle_separated_map_string_string.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flag
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// LangleSeparatedMapStringString can be set from the CLI with "<" instead of "="
+// as the key-value separator. Pairs are still separated with ",".
+type LangleSeparatedMapStringString map[string]string
+
+// String implements github.com/spf13/pflag.Value
+func (m LangleSeparatedMapStringString) String() string {
+	pairs := []string{}
+	for k, v := range m {
+		pairs = append(pairs, fmt.Sprintf("%s<%s", k, v))
+	}
+	sort.Strings(pairs)
+	return strings.Join(pairs, ",")
+}
+
+// Set implements github.com/spf13/pflag.Value
+func (m LangleSeparatedMapStringString) Set(value string) error {
+	// clear old values
+	for k := range m {
+		delete(m, k)
+	}
+	for _, s := range strings.Split(value, ",") {
+		if len(s) == 0 {
+			continue
+		}
+		arr := strings.SplitN(s, "<", 2)
+		if len(arr) != 2 {
+			return fmt.Errorf("malformed pair, expect string<string")
+		}
+		k := strings.TrimSpace(arr[0])
+		v := strings.TrimSpace(arr[1])
```

Comment:
Was this space trimming done before? If not, I prefer to avoid normalization since it limits expressable values in a generic flag var parser

Reply:
No, it wasn't. I was trying to be consistent with the ConfigurationMap shim. Wouldn't a user expect something like `--flag "a < 1, b < 2"` to produce `map[string]string{"a": "1", "b": "2"}`, rather than `map[string]string{"a ": " 1 ", " b ": " 2"}`, where they don't target their intended keys, and values may be malformed depending on their interpretation? 

I understand that this prevents us from using prefix/postfix whitespace, but I'd say that none of our APIs should depend on prefix/postfix whitespace anyway, because it's very confusing to read and easy to mess up (is that arbitrary-width bit of nothingness one space or two?).

Reply:
I can buy that, and it becomes moot once config is specified via structured files anyway

Reply:
Right, that too.